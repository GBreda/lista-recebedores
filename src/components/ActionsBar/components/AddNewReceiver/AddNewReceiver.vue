<template>
  <div class="add-new-receiver" data-test-id="add-new-receiver" @click="addNewReceiver">
    <span class="add-new-receiver__text">Seus favorecidos</span>
    <img class="add-new-receiver__icon" src="@/assets/images/cross-icon.png" />
  </div>
</template>

<script setup>
import { useRouter } from 'vue-router'

const router = useRouter()

const addNewReceiver = () => {
  router.push({ name: 'add-new-receiver' })
}
</script>

<style lang="scss" scoped>
.add-new-receiver {
  display: flex;
  justify-content: space-between;
  align-items: center;
  cursor: pointer;

  &__text {
    font-weight: 300;
    font-size: 2.8rem;
  }

  &__icon {
    padding: 1rem;
    border-radius: 50%;
    background-color: $green;
    margin-left: 1rem;
  }
}
</style>
