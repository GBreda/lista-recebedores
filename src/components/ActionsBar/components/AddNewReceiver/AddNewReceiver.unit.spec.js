import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import AddNewReceiver from './AddNewReceiver.vue'

describe('AddNewReceiver', () => {
  let wrapper

  it('should mount the component', () => {
    wrapper = mount(AddNewReceiver)

    expect(wrapper.vm).toBeDefined()
  })

  it('should match the snapshot', () => {
    wrapper = mount(AddNewReceiver)

    expect(wrapper.element).toMatchSnapshot()
  })

  it('should emit event to add new receiver', async () => {
    wrapper = mount(AddNewReceiver)

    wrapper.find("[data-test-id='add-new-receiver']").trigger('click')

    expect(wrapper.emitted()).toHaveProperty('add:new-receiver')
  })
})
