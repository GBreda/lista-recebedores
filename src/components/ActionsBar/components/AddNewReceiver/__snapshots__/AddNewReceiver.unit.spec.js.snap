// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddNewReceiver > should match the snapshot 1`] = `
<div
  class="add-new-receiver"
  data-test-id="add-new-receiver"
  data-v-25b32079=""
>
  <span
    class="add-new-receiver__text"
    data-v-25b32079=""
  >
    Seus favorecidos
  </span>
  <img
    class="add-new-receiver__icon"
    data-v-25b32079=""
    src="/src/assets/images/cross-icon.png"
  />
</div>
`;
